src/diffusers/models/modeling_utils.py

```diff
@@ -314,17 +326,68 @@ def save_pretrained(
         state_dict = model_to_save.state_dict()
 
         weights_name = SAFETENSORS_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
-        weights_name = _add_variant(weights_name, variant)
+        weights_name = _add_variant(weights_name, variant, add_suffix_keyword=True)
 
         # Save the model
-        if safe_serialization:
-            safetensors.torch.save_file(
-                state_dict, Path(save_directory, weights_name).as_posix(), metadata={"format": "pt"}
+        state_dict_split = None
+        if is_huggingface_hub_version(">=", "0.23.2"):
+            state_dict_split = split_torch_state_dict_into_shards(
+                state_dict, max_shard_size=max_shard_size, filename_pattern=weights_name
             )
-        else:
-            torch.save(state_dict, Path(save_directory, weights_name).as_posix())
+        # Clean the folder from a previous save
+        if is_main_process:
+            for filename in os.listdir(save_directory):
+                if filename in state_dict_split.filename_to_tensors.keys():
+                    continue
+                full_filename = os.path.join(save_directory, filename)
+                if not os.path.isfile(full_filename):
+                    continue
+                else:
+                    weights_without_ext = weights_name.replace(".bin", "").replace(".safetensors", "")
+                    filename_without_ext = filename.replace(".bin", "").replace(".safetensors", "")
+                    # make sure that file to be deleted matches format of sharded file, e.g. pytorch_model-00001-of-00005
+                    if filename.startswith(weights_without_ext) and re.fullmatch(filename_without_ext) is not None:
+                        os.remove(full_filename)
 
-        logger.info(f"Model weights saved in {Path(save_directory, weights_name).as_posix()}")
+        # Save the model
+        if state_dict_split is not None:
```

Comment:
The might require folks to update `huggingface_hub` just to avail a single workflow of handling model serialization. FWIW, model serialization doesn't only take place in training workflows. 

What if a user wants to just do:

```python
...

model = UNet2DConditionModel.from_pretrained(...)
pipeline = DiffusionPipeline.from_pretrained(..., unet=model)
pipeline.save_pretrained(...)
```

If their `huggingface_hub` installation isn't updated, then this would fail and IMO this is not a good developer experience. But I am happy to stand corrected if this is not the right way to think about it.